src/components/content/credential/Credential.tsx

```diff
@@ -0,0 +1,228 @@
+                                Details
+                            </Button>
+                            <Popconfirm
+                                title='Delete the Credential'
+                                description='Are you sure to delete the Credential?'
+                                okText='Yes'
+                                cancelText='No'
+                                onConfirm={() => deleteCredential(record)}
+                            >
+                                <Button type='primary' icon={<MinusCircleOutlined />}>
+                                    delete
+                                </Button>
+                            </Popconfirm>
+                            <Button
+                                type='primary'
+                                icon={<InfoCircleOutlined />}
+                                onClick={() => updateCredential(record)}
+                            >
+                                update
+                            </Button>
+                        </Space>
+                    </>
+                );
+            },
+        },
+    ];
+
+    const addCredential = () => {
+        setIsAddOpen(true);
+    };
+
+    const deleteCredential = (record: CredentialVariables) => {
+        void CredentialsManagementService.deleteCredential(record.csp, record.xpanseUser, record.type)
+            .then(() => {
+                getTipInfo('success', 'delete success!');
+                getCredentials();
+            })
+            .catch(() => {
+                getTipInfo('error', 'delete failed!');
+            });
+    };
+
+    const updateCredential = (record: CredentialVariables) => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const createCredential: CreateCredential = JSON.parse(JSON.stringify(record));
+        createCredential.timeToLive = 3600;
+        const userName: string | null = localStorage.getItem(usernameKey);
+        if (userName === null) {
+            return;
+        }
+        createCredential.xpanseUser = userName;
+        setCreateCredential(createCredential);
+        setIsUpdateOpen(true);
+    };
+
+    const details = (record: CredentialVariables) => {
+        setIsDetailsOpen(true);
+        setCredentialDetails(record.variables);
+    };
+
+    function getCredentials(): void {
+        const userName: string | null = localStorage.getItem(usernameKey);
+        if (!userName) {
+            return;
+        }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+        void CredentialsManagementService.getCredentialsByUser(userName).then((resp) => {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            if (resp.length > 0) {
+                setCredentialDefinitionList(resp as CredentialVariables[]);
```

Comment:
do we need the as CredentialVariables[]? resp: CredentialVariables[]

we must also rename resp to credentialVariables. 
Also we must see how to remove these errors which you hve suppressed. They seem to be valid warnings.